src/kademlia/routing_table.cpp

```diff
@@ -626,6 +631,7 @@ routing_table::add_node_status_t routing_table::add_node_impl(node_entry e)
 	// split the last bucket
 	bool const can_split = (std::next(i) == m_buckets.end()
 		&& m_buckets.size() < 159)
+		&& e.verified
```

Comment:
This is probably fine, but it feels wrong for the contents of the routing table to depend on the order in which nodes enter it. Maybe set a threshold based on the number of verified nodes in the last bucket?

Reply:
yeah, I think I will let this patch sit for a while. I won't block cutting 1.2. The simulation tests also need to be updated

Reply:
you mean that whether the last node (causing the bucket to split) is verified or not, should not matter so much as how many verified nodes there are in the bucket? that kind of makes sense. Although, you would imagine there would be a verified node eventually that would come around to causing the split.

Reply:
Right. With the way this PR is now a bucket full of unverified nodes can be split by a single verified node, but a bucket full of verified nodes will not be split when an unverified node is added. So an attacker only needs to come up with a single verified ID in the appropriate range to force a split. This might let them trigger marginally more splits for a given number of attacker controlled IPs than if they needed to insert, say, `K / 2` verified peers in the home bucket before it will be split.